plugin/server/config_service.py

```
@@ -0,0 +1,620 @@
+    with _config_update_locks_lock:
+        if plugin_id not in _config_update_locks:
+            _config_update_locks[plugin_id] = threading.Lock()
+        lock = _config_update_locks[plugin_id]
+
+    with lock:
+        config_path = get_plugin_config_path(plugin_id)
+
+        try:
+            parsed_new = tomllib.loads(toml_text or "")
+        except Exception as e:
+            raise HTTPException(status_code=400, detail=f"Invalid TOML format: {str(e)}") from e
```

Ruff (0.14.10)

Use explicit conversion flag

Replace with conversion flag

(RUF010)